plugins/org.jboss.tools.openshift.common.core/src/org/jboss/tools/openshift/common/core/utils/URIUtils.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors: Red Hat, Inc.
+ ******************************************************************************/
+package org.jboss.tools.openshift.common.core.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Helper methods for manipulating URIs
+ */
+public class URIUtils {
+	
+	private URIUtils(){
+	}
+	
+	public static  Map<String, String> splitFragment(String location){
+		URI uri = null;
+		try {
+			uri = new URI(location);
+		} catch (URISyntaxException e) {
+			throw new RuntimeException(e);
+		}
+		return splitFragment(uri);
+	}
+	
+	public static Map<String, String> splitFragment(URI uri){
```

Comment:
can you please explain this? in code it looks much more as if you would split argument (key/value-) pairs than fragments: http://en.wikipedia.org/wiki/Fragment_identifier